File: snippets/fsharp/System/Type/GetConstructors/source1.fs
module source1

// <Snippet1>
type t() =
    static do ()
    new(i: int) = t ()

let p = typeof<t>.GetConstructors()
printfn $"{p.Length}"

for c in p do
    printfn $"{c.IsStatic}"
// </Snippet1>

File: snippets/csharp/System/Type/GetConstructors/source1.cs
// <Snippet1>
using System;
using System.Reflection;

public class ConstructorSample1
{
    public ConstructorSample1() { }
    static ConstructorSample1() { }
    public ConstructorSample1(int i) { }
    public static void Run()
    {
        ConstructorInfo[] p = typeof(ConstructorSample1).GetConstructors();
        Console.WriteLine(p.Length);

        for (int i = 0; i < p.Length; i++)
        {
            Console.WriteLine(p[i].IsStatic);
        }
    }
}
// </Snippet1>
